File: Java/dependency-injection/control-freak/src/main/java/com/github/grzesiek_galezowski/DependencyInjection/_1_ControlFreak/Outbound/TcpSocket.java
package com.github.grzesiek_galezowski.DependencyInjection._1_ControlFreak.Outbound;

public class TcpSocket {
  public void open() {
    System.out.println("open");
  }

  public void close() {
    System.out.println("closing");
  }

  public void send(String lol) {
    System.out.println(lol);
  }
}

File: CSharp/DependencyInjection/DependencyInjection/_1_ControlFreak/Outbound/TcpSocket.cs
using System;

namespace DependencyInjectionBefore._1_ControlFreak.Outbound
{
  class TcpSocket
  {
    public void Open()
    {
      Console.WriteLine("open");
    }

    public void Close()
    {
      Console.WriteLine("closing");
    }

    public void Send(string lol)
    {
      Console.WriteLine(lol);
    }
  }
}